aws/static_provider.go

```diff
@@ -31,7 +33,13 @@ func NewStaticCredentialsProvider(key, secret, session string) StaticCredentials
 }
 
 // Retrieve returns the credentials or error if the credentials are invalid.
-func (s StaticCredentialsProvider) Retrieve() (Credentials, error) {
+func (s StaticCredentialsProvider) Retrieve(ctx context.Context) (Credentials, error) {
+	select {
+	case <-ctx.Done():
+		return Credentials{}, awserr.New(ErrCodeRequestCanceled, "context canceled", ctx.Err())
+	default: // do nothing
+	}
+
```

Comment:
Probably don't need this check, the context can probably be ignored in this case. If the context is canceled the request will be terminated before it is invoked.

Reply:
Updated in #461